packages/bull-vault/src/FlashBull.sol

```diff
@@ -86,15 +85,14 @@ contract FlashBull is UniBull {
 
     event FlashWithdraw(uint256 bullAmount);
 
-    constructor(address _bull, address _factory, address _oracle) UniBull(_factory) {
+    constructor(address _bull, address _factory) UniFlash(_factory) {
         bullStrategy = _bull;
-        oracle = _oracle;
         crab = IBullStrategy(_bull).crab();
-        wPowerPerp = IController(IBullStrategy(bullStrategy).powerTokenController()).wPowerPerp();
-        weth = IController(IBullStrategy(bullStrategy).powerTokenController()).weth();
-        usdc = IController(IBullStrategy(bullStrategy).powerTokenController()).quoteCurrency();
-        ethWSqueethPool = IController(IBullStrategy(bullStrategy).powerTokenController()).wPowerPerpPool();
-        ethUSDCPool = IController(IBullStrategy(bullStrategy).powerTokenController()).ethQuoteCurrencyPool();
+        wPowerPerp = IController(IBullStrategy(_bull).powerTokenController()).wPowerPerp();
+        weth = IController(IBullStrategy(_bull).powerTokenController()).weth();
```

Comment:
Maybe we should set a var for the powerTokenController so we don't have to constantly call this function? Which is cheaper?